### PR TITLE
Reorganize content pertaining to dbt Cloud jobs, add info about Artifacts Job

### PR DESCRIPTION
**Pull request by leoebfolsom**
I realized some of the content in [Integrations > Orchestration > dbt Cloud](https://docs.datafold.com/integrations/orchestration/dbt_cloud/prerequisites) was completely duplicative of content in [Guides > CI > dbt Cloud](https://docs.datafold.com/guides/ci_guides/dbt_cloud#prerequisites). (Click on the links to see what I mean.)

I moved the information that is necessary to a user getting up and running with dbt Cloud to Integrations > Orchestration > dbt Cloud since that is the section that is organized by the sections of the Datafold app's settings > configurations, and likely what a user would look to when getting started. Of course, I also removed duplicative information.

I also removed the instructions for creating PR and Prod dbt Cloud jobs, since this table stakes prereq for having dbt Cloud CI/CD at all, and dbt Labs provides good documentation.

Based on discussion with Will and Dan, I moved the Merge Trigger instructions back to Guides, since that's more advanced and not strictly necessary to get value out of Datafold--many customers get a lot of value out of the Artifacts job; there are tradeoffs.

Finally, I added some information about the dbt artifacts job, which is necessary for users who do not create a merge trigger job.

### CODE REVIEW

**Comment by leoebfolsom** [comment]
Please review with these notes in mind:

- For some reason, images aren't rendering properly for me when I `npm run build && npm run serve`. I suspect this is a docusaurus caching issue--please see if the images on `integrations/orchestration/dbt_cloud/prerequisites` load properly for you.
- I realize we are planning a larger reorg of the docs, but I thought these issues were important enough to include in a quick fix to support current onboarding efforts.

**Comment by williebsweet** [comment]
@leoebfolsom Not a caching issue. I think it's the breaks in the code affecting the image paths.

**Comment by leoebfolsom** [comment]
@williebsweet fixed the images 🙌

**Comment by dlawin** [comment]
yep they're loading for me https://deploy-preview-107--datafold-docs.netlify.app/integrations/orchestration/dbt_cloud/prerequisites

File: docs/guides/ci_guides/slim_diff.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ models:
   <project_name>:
     <directory_name>:        
       +materialized: view
-      <model_name>:
+      - name: <model_name>:
```
Comment (dlawin) [inline docs/guides/ci_guides/slim_diff.md:68]:
```suggestion
      - name: <model_name>
```

**Review by dlawin — APPROVED** [review]
I dig it

File: docs/guides/ci_guides/slim_diff.md (hunk 1 above)
Comment (leoebfolsom) [reply docs/guides/ci_guides/slim_diff.md:68]: Updated in separate PR 🙌  https://github.com/datafold/datafold-docs/pull/121

**Pull request merged**